### DISCUSSION TITLE
Interactive layout and page with if-else

### DISCUSSION THREAD

**Discussion by b-avb** [nuxt/nuxt#17757] [Questions]
I have two layouts, backout and blank, and I have a simple page with a span and a form. I'm trying to change the layouts and show the span and hide the form when it's submitted. I don't get this result, if I click to submit only the layout is changed, but the span isn't shown and the form isn't hidden, only work when I click again to submit.

layouts/backout.vue
```html
<template>
  <div class="page">
    <h2>My backout</h2>
    <slot />
  </div>
</template>
```

The layout blanck only have the slot.

layouts/blanck.vue
```html
layouts/backout.vue
<template>
  <div class="page">
    <slot />
  </div>
</template>
```

pages/mypage.vue
```html
<template>
  <span v-if="submitted">Hello world</span>
  <form v-else @submit.prevent="handleSubmit">
    <input type="text">
    <button type="submit">Show/Hide</button>
  </form>
</template>

<script lang="ts" setup>
const route = useRoute();
const submitted = ref<false>;

definePageMeta({
  layout: "backout",
});

function handleSubmit(){
  submitted.value = true;
  route.meta.layout = "blanck";
}
</script>
```

**Comment by danielroe** [comment]
You need to have a single root node in your layouts and pages.

**Reply by b-avb** [reply]
Fixed using the [manual control](https://v3.nuxtjs.org/docs/directory-structure/layouts/#example-manual-control-with-pages) and passing the name as a variable.

```html
<template>
  <NuxtLayout :name="layoutName">
    <span v-if="submitted">Hello world</span>
    <form v-else @submit.prevent="handleSubmit">
      <input type="text">
      <button type="submit">Show/Hide</button>
    </form>
  <NuxtLayout :name="submitted ? 'default' : 'backout'">
</template>

<script lang="ts" setup>
const route = useRoute();
const submitted = ref<boolean>(false);
const layoutName = ref<string>('blanck');

definePageMeta({
  layout: false,
});

function handleSubmit(){
  submitted.value = true;
  layoutName.value = "backout";
}
</script>
```
Finally, I updated the name `blanck` to `default` to follow the naming convention for layouts.
